src/L1/rollup/ScrollChain.sol

```diff
@@ -438,149 +369,67 @@ contract ScrollChain is OwnableUpgradeable, PausableUpgradeable, IScrollChain {
-        );
-
-        _afterFinalizeBatch(_totalL1MessagesPoppedOverall, _batchIndex, _batchHash, _postStateRoot, _withdrawRoot);
-    }
-    */
-
-    /// @inheritdoc IScrollChain
-    /// @dev Memory layout of `_blobDataProof`:
-    /// ```text
-    /// | z       | y       | kzg_commitment | kzg_proof |
-    /// |---------|---------|----------------|-----------|
-    /// | bytes32 | bytes32 | bytes48        | bytes48   |
-    /// ```
-    function finalizeBatchWithProof4844(
-        bytes calldata _batchHeader,
-        bytes32, /*_prevStateRoot*/
-        bytes32 _postStateRoot,
-        bytes32 _withdrawRoot,
-        bytes calldata _blobDataProof,
-        bytes calldata _aggrProof
-    ) external override OnlyProver whenNotPaused {
-        (uint256 batchPtr, bytes32 _batchHash, uint256 _batchIndex) = _beforeFinalizeBatch(
-            _batchHeader,
-            _postStateRoot
-        );
-
-        // compute public input hash
-        bytes32 _publicInputHash;
-        {
-            bytes32 _dataHash = BatchHeaderV0Codec.getDataHash(batchPtr);
-            bytes32 _blobVersionedHash = BatchHeaderV1Codec.getBlobVersionedHash(batchPtr);
-            bytes32 _prevStateRoot = finalizedStateRoots[_batchIndex - 1];
-            // verify blob versioned hash
-            _checkBlobVersionedHash(_blobVersionedHash, _blobDataProof);
-            _publicInputHash = keccak256(
-                abi.encodePacked(
-                    layer2ChainId,
-                    _prevStateRoot,
-                    _postStateRoot,
-                    _withdrawRoot,
-                    _dataHash,
-                    _blobDataProof[0:64],
-                    _blobVersionedHash
-                )
-            );
-        }
-
-        // load version from batch header, it is always the first byte.
-        uint256 batchVersion = BatchHeaderV0Codec.getVersion(batchPtr);
-        // verify batch
-        IRollupVerifier(verifier).verifyAggregateProof(batchVersion, _batchIndex, _aggrProof, _publicInputHash);
-
-        // Pop finalized and non-skipped message from L1MessageQueue.
-        uint256 _totalL1MessagesPoppedOverall = BatchHeaderV0Codec.getTotalL1MessagePopped(batchPtr);
-        _popL1MessagesMemory(
-            BatchHeaderV1Codec.getSkippedBitmapPtr(batchPtr),
-            _totalL1MessagesPoppedOverall,
-            BatchHeaderV0Codec.getL1MessagePopped(batchPtr)
-        );
-
-        _afterFinalizeBatch(_totalL1MessagesPoppedOverall, _batchIndex, _batchHash, _postStateRoot, _withdrawRoot);
-    }
-
     /// @inheritdoc IScrollChain
+    /// @dev All batches in the given bundle should have the same version and version <= 4 or version >= 6.
     function finalizeBundleWithProof(
-        bytes calldata _batchHeader,
-        bytes32 _postStateRoot,
-        bytes32 _withdrawRoot,
-        bytes calldata _aggrProof
+        bytes calldata batchHeader,
+        bytes32 postStateRoot,
+        bytes32 withdrawRoot,
+        bytes calldata aggrProof
```

Comment:
Just curious, why did you change this? Previously you consistently preferred underscore for function arguments.